mods/beds/functions.lua

```diff
@@ -7,17 +7,42 @@ if enable_respawn == nil then
 end
 
 -- Helper functions
+local facedir_types = {
+	colorwallmounted = 8, -- 3 bits are used for facedir
+	colorfacedir = 32     -- 5 bits are used for facedir
+}
 
 local function get_look_yaw(pos)
 	local n = minetest.get_node(pos)
-	if n.param2 == 1 then
-		return pi / 2, n.param2
-	elseif n.param2 == 3 then
-		return -pi / 2, n.param2
-	elseif n.param2 == 0 then
-		return pi, n.param2
+	local face_dir = n.param2
+	if face_dir < 0 or face_dir > 24 then
```

Comment:
facedir will never be less than 0.
'colorwallmounted' will result in facedir 9+ so only running this code when facedir > 24 is wrong.
Since both colorwallmounted and facedir will only ever use 0-3 for the rotation component of a bed it's best to simply strip off the first 6 bits.

Sorry, looks like this is what your 1st commit did.